frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/contextmenu/SyncSelectedToParentByMergeAction.java

```diff
@@ -1,16 +1,50 @@
 package com.virtuslab.gitmachete.frontend.actions.contextmenu;
 
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import com.intellij.ide.util.PropertiesComponent;
 import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.ui.MessageDialogBuilder;
+import com.intellij.openapi.ui.Messages;
 import io.vavr.control.Option;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
 
 import com.virtuslab.gitmachete.frontend.actions.base.BaseSyncToParentByMergeAction;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.WarnAboutSyncToParentByMergeDialog;
 import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeySelectedBranchName;
 
 public class SyncSelectedToParentByMergeAction extends BaseSyncToParentByMergeAction
     implements
       IExpectsKeySelectedBranchName {
+
+  public static final String SHOW_MERGE_WARNING = "git-machete.merge.warning.show-again";
```

Comment:
I'd skip `again`, as the first-ever occurrence is also (theoretically) controlled by this config key... just:
```suggestion
  public static final String SHOW_MERGE_WARNING = "git-machete.merge.warning.show";
```
Same for Reset